### PR TITLE
clarify handling of missing or invalid data

### PR DESCRIPTION
**Pull request by mmcnl**

### CODE REVIEW

File: src/histogram-date-range.ts [hunk 1]
```diff
@@ -122,6 +124,17 @@ export class HistogramDateRange extends LitElement {
     this.requestUpdate();
   }
 
+  private get hasValidData(): boolean {
+    return Boolean(
+      this.data.minDate &&
+        dayjs(this.data.minDate).valueOf() &&
+        this.data.maxDate &&
+        dayjs(this.data.maxDate).valueOf() &&
+        Array.isArray(this.data.bins) &&
```
Comment (jbuckner) [inline src/histogram-date-range.ts:133]: What other type could `this.data.bins` be?

File: src/histogram-date-range.ts [hunk 2]
```diff
@@ -38,10 +39,10 @@ const tooltipFontSize = css`var(--histogramDateRangeTooltipFontSize, 1.1rem)`;
...
 type SliderId = 'slider-min' | 'slider-max';
 
-interface HistogramInputData {
-  minDate: string;
-  maxDate: string;
-  bins: number[];
+class HistogramInputData {
+  minDate = '';
+  maxDate = '';
+  bins: number[] = [];
```
Comment (jbuckner) [inline src/histogram-date-range.ts:45]: It looks like you're checking for the existence of these properties down below, does that mean they can be optional? If so, I would mark them optional so typescript can enforce them.

File: src/histogram-date-range.ts (hunk 1 above)
Comment (mmcnl) [reply src/histogram-date-range.ts:133]: if someone passed in a malformed data object, eg `data.bins = "xyz"` I wanted it to no blow up when I try to `map` that property later on

File: src/histogram-date-range.ts (hunk 2 above)
Comment (mmcnl) [reply src/histogram-date-range.ts:45]: they would only be missing if someone passed in an invalid data object. (see the test file for examples). but if we don't have valid data, then we don't try to access `minDate` etc so I didn't want to make them optional. Does this make sense?

**Review by jbuckner — APPROVED** [review]
:shipit:

**Pull request merged**